src/main/java/com/iota/iri/IRI.java

```diff
@@ -16,314 +16,358 @@
-        iota = new Iota(configuration);
-        ixi = new IXI(iota);
-        api = new API(iota, ixi);
-        shutdownHook();
-
-        if (configuration.booling(DefaultConfSettings.DEBUG)) {
-            log.info("You have set the debug flag. To enable debug output, you need to uncomment the DEBUG appender in the source tree at iri/src/main/resources/logback.xml and re-package iri.jar");
+
+    public static void main(String[] args) throws IOException {
+        // Logging is configured first before any references to Logger or LoggerFactory.
+        // Any public method or field accessors needed in IRI should be put in IRI and then delegate to IRILauncher. That
+        // ensures that future code does not need to know about this setup.
+        configureLogging();
+        IRILauncher.main(args);
+    }
+
+    private static void configureLogging() {
+        String config = System.getProperty("logback.configurationFile");
+        String level = System.getProperty("logging-level", "").toUpperCase();
+        switch (level) {
+            case "OFF":
+            case "ERROR":
+            case "WARN":
+            case "INFO":
+                break;
+            case "DEBUG":
+            case "TRACE":
+            case "ALL":
+                level = "DEBUG";
+                break;
+            default:
+                level = "INFO";
+        }
+        System.getProperties().put("logging-level", level);
+        System.out.println("Logging: property 'logging-level' set to : [" + level + "]");
+        System.out.println("Logging: levels can be one of: OFF, ERROR, WARN, INFO, DEBUG");
+        if (config != null) {
+            System.out.println("Logging: alternate logging configuration file specified at: '" + config + "'");
         }
+    }
 
-        if (configuration.booling(DefaultConfSettings.EXPORT)) {
-            File exportDir = new File("export");
-            // if the directory does not exist, create it
-            if (!exportDir.exists()) {
-                log.info("Create directory 'export'");
-                try {
-                    exportDir.mkdir();
-                } catch (SecurityException e) {
-                    log.error("Could not create directory", e);
-                }
+    private static class IRILauncher {
+        private static final Logger log = LoggerFactory.getLogger(IRILauncher.class);
+
+        public static Iota iota;
+        public static API api;
+        public static IXI ixi;
+        public static Configuration configuration;
+
+        private static final String TESTNET_FLAG_REQUIRED = "--testnet flag must be turned on to use ";
+
+        public static void main(final String[] args) throws IOException {
+            configuration = new Configuration();
+
+            if (!isValidated(configuration, args)) {
+                printUsage();
+                return;
```

Comment:
(0) return is a normal non-error return.
No need to use forbidden J2EE/JEE System.exit(0) here.
The only one that returns a -2, now throws instead.